src/lambda/gc01_check_alerts_flag_misuse/app.py

```diff
@@ -0,0 +1,418 @@
+            ex.response["Error"]["Code"] = "InternalError"
+        raise ex
+
+def rule_event_pattern_matches_guard_duty_findings(rule_event_pattern:str | None):
+    if rule_event_pattern == None:
+        return False
+    logger.info("rule_event_pattern: %s", rule_event_pattern)
+    
+    event_pattern_dict = json.loads(rule_event_pattern)
+    logger.info("event_pattern_dict: %s", event_pattern_dict)
+    return "aws.guardduty" in event_pattern_dict.get("source", []) and "GuardDuty Finding" in event_pattern_dict.get("detail-type", [])
+
+def check_rules_sns_target_is_setup(rules, event):
+    for rule in rules:
+        logger.info("Checking rule: %s", rule)
+        if rule.get("State") == "DISABLED":
+            return build_evaluation(
+                event["accountId"],
+                "NON_COMPLIANT",
+                event,
+                resource_type="AWS::Events::Rule",
+                annotation="Rule is disabled.",
+            )   
+        
+        rule_name = rule.get("Name")          
+        targets = fetch_rule_targets(rule_name)
+
+        for target in targets:
+            logger.info("Checking rule target: %s", target)
+            # is target an SNS input transformer?
+            target_arn: str = target.get("Arn")
+            if target_arn.startswith("arn:aws:sns:") :
+                # yes, get a list of topic subscriptions
+                subscriptions =  get_topic_subscriptions(target_arn)
+                # then search topic for a subscription with "email" protocol and is confirmed
+                for subscription in subscriptions:
+                    logger.info("Checking target subscriptions: %s", subscription)
+                    if subscription.get("Protocol") == "email":
+                        subscription_arn = subscription.get("SubscriptionArn")
+                        if subscription_is_confirmed(subscription_arn):
+                            return build_evaluation(
+                                event["accountId"],
+                                "COMPLIANT",
+                                event,
+                                resource_type=DEFAULT_RESOURCE_TYPE,
+                                annotation="An Event rule that has a SNS topic and subscription to send notification emails setup and confirmed.",
+                            )
+    
+    return build_evaluation(
+        event["accountId"],
+        "NON_COMPLIANT",
+        event,
+        resource_type=DEFAULT_RESOURCE_TYPE,
+        annotation="An Event rule that has a SNS topic and subscription to send notification emails is not setup or confirmed.",
+    )         
+
+def get_rule_naming_convention(rule_naming_convention_file_path):
+    bucket, key = extract_bucket_name_and_key(rule_naming_convention_file_path)
+    response = AWS_S3_CLIENT.get_object(Bucket=bucket, Key=key)
+    rule_naming_convention = response.get("Body").read().decode("utf-8")
+    return rule_naming_convention
```

Comment:
With this current implementation I am retrieving the naming covention by reading a file in an s3 bucket. I'm doing this because Conformance Pack inputs can't have special characters except for - and _ , which would limit the regex a user could use to define the naming convention. However if the naming convention should just be a string prefix then I can change it to just be a string parameter. Leaving this comment to open it up for discussion.